src/main/java/org/isf/telemetry/gui/TelemetryGUI.java

```diff
@@ -0,0 +1,244 @@
+			result.add(wrapper);
+		});
+
+		return result;
+	}
+
+	private void fireTelemetryInserted(Telemetry telemetry) {
+		AWTEvent event = new AWTEvent(telemetry, AWTEvent.RESERVED_ID_MAX + 1) {
+
+			private static final long serialVersionUID = 1L;
+		};
+
+		EventListener[] listeners = telemetryListeners.getListeners(TelemetryListener.class);
+		for (EventListener listener : listeners) {
+			((TelemetryListener) listener).telemetryInserted(event);
+		}
+	}
+
+	/**
+	 * Action for confirmation button: inserts for the first time in the telemetry table or updates the existing row
+	 * 
+	 * @param permissions
+	 * @param telemetryManager
+	 * @return
+	 */
+	private ActionListener buildConfirmationActionListener(List<CheckBoxWrapper> checkboxes, TelemetryManager telemetryManager, TelemetryUtils telemetryUtils) {
+		return new ActionListener() {
+
+			public void actionPerformed(ActionEvent e) {
+				Map<String, Boolean> consentMap = buildConsentData(checkboxes);
+				if (this.isReallyEnabled(consentMap)) {
+					Telemetry telemetry = telemetryManager.enable(consentMap);
+					fireTelemetryInserted(telemetry);
+					try {
+						GeneralData.initialize();
+						telemetryUtils.sendTelemetryData(consentMap, GeneralData.DEBUG);
+					} catch (RuntimeException | OHException f) {
+						LOGGER.error("Something strange happened: " + f.getMessage());
+						LOGGER.error(ExceptionUtils.retrieveExceptionStacktrace(f));
+					}
+				} else {
+					telemetryManager.disable(new HashMap<>());
+				}
+				removeTelemetryListener(parent);
+				dispose();
+			}
+
+			private void removeTelemetryListener(TelemetryListener listener) {
+				telemetryListeners.remove(TelemetryListener.class, listener);
+			}
+
+			private boolean isReallyEnabled(Map<String, Boolean> cd) {
+				if (cd == null || cd.isEmpty()) {
+					return false;
+				}
+				return cd.keySet().stream().anyMatch(key -> Boolean.TRUE.equals(cd.get(key)));
+			}
+
+			private Map<String, Boolean> buildConsentData(List<CheckBoxWrapper> permissions) {
+
+				Map<String, Boolean> result = new HashMap<>();
+				permissions.forEach(per -> {
+					result.put(per.getId(), Boolean.valueOf(per.getCheckbox().isSelected()));
+				});
+
+				return result;
+			}
+
+		};
+	}
+
+	private ActionListener buildCancelButtonActionListener(TelemetryManager telemetryManager) {
+		return new ActionListener() {
+
+			public void actionPerformed(ActionEvent e) {
+				dispose();
+			}
+		};
+
+	}
+
+	public JPanel panel(List<CheckBoxWrapper> checkboxes, JButton confirmButton, JButton cancelButton) {
+
+		JPanel panel = new JPanel(new SpringLayout());
+
+		panel.add(new JLabel(MessageBundle.getMessage("angal.common.userid.label")));
+		checkboxes.forEach(chb -> {
+			panel.add(chb.getCheckbox());
+		});
+		SpringUtilities.makeCompactGrid(panel, checkboxes.size(), 1, 5, 5, 5, 5);
+
+		JPanel buttons = new JPanel();
+		buttons.setLayout(new FlowLayout());
+		buttons.add(confirmButton);
+		buttons.add(cancelButton);
```

Comment:
In my opinion buttons should be:
- Confirm (as it is now)
- Ask me later (it will ask next time)
- Never ask again (it disable the feature)

Reply:
done